core/src/main/resources/bosatsu/predef.bosatsu

```diff
@@ -88,6 +97,15 @@ enum Option:
   None
   Some(v)
 
+struct Order(fn: a -> a -> Comparison)
```

Comment:
not really sure if we should name this type or not, but you can write combinators on it, so I tend to say yes.

Eventually we could optimize 1 item structs to not box. I think that's a good idea for sure for a compiled backend.